src/App.js

```diff
@@ -1,17 +1,56 @@
+// import the required libraries
 import React from 'react';
 import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+
+// import the pages/components
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware, compose } from 'redux';
+import thunk from 'redux-thunk'; // import thunk middleware
+import createSagaMiddleware from '@redux-saga/core';
 import { LandingPage } from './components/landingPage/components';
-import LoginPage from './components/login/components';
+import { components as LoginPage } from './components/login';
 import AdminLoginPage from './components/adminLoginPage/components';
 
+// import the combinedSagas and the combinedReducers
+import rootSaga from './sagas';
+import allReducers from './reducers'; // import all the reducers
+
+// import the modules for making the store
+
+// initialise the saga middleware
+const sagaMiddleware = createSagaMiddleware();
+
+// initialise an array of all the middlewares we are going to use
+const middleware = [thunk, sagaMiddleware];
+const initialState = {};
+
+// create the globa store
+const store = createStore(
+    allReducers,
+    initialState,
+    // wrap all of them in a compose in order to wrap them as a single argument
+    compose(
+        applyMiddleware(...middleware)
+        // window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
+    )
+);
+
+// run the middleware for the rootsaga
+sagaMiddleware.run(rootSaga);
+
+// the exported app
 const App = () => (
-    <Router>
-        <Switch>
-            <Route exact path="/" component={LandingPage} />
-            <Route exact path="/login" component={LoginPage} />
-            <Route exact path="/admin" component={AdminLoginPage} />
-        </Switch>
-    </Router>
+    <Provider store={store}>
+
+        <Router>
+            <Switch>
+                <Route exact path="/" component={LandingPage} />
+                <Route exact path="/login" component={LoginPage} />
+                <Route exact path="/admin" component={AdminLoginPage} />
+            </Switch>
+        </Router>
```

Comment:
you want your code to be loosely coupled, separate the route to its own component

Reply:
i will do this, thanks